### PR TITLE
Suvrath

### PR DESCRIPTION
**Pull request by suvrathpen**
Updated Backend code, moving simple schema to tasks.js, moving helper functions to dashboard.js etc

### CODE REVIEW

File: imports/api/tasks/tasks.js [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+  name: { type: String },
+  completed_on: {type: Date, optional: true},
+  created_on: {type: Date},
+  due_on: { type: Date },
+  is_urgent: { type: Boolean},
+  is_important: { type: Boolean}
+
+});
+
+Tasks.attachSchema(Schemas.Tasks);
```
Comment (Ligerx) [inline imports/api/tasks/tasks.js]: need to fix this.
Remove 'export' from line 3. At the bottom of the file, type export Tasks

File: imports/api/tasks/tasks.js [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
 
 export const Tasks = new Mongo.Collection('tasks');
+
+var Schemas = {};
+
+Schemas.Tasks = new SimpleSchema({
+  name: { type: String },
+  completed_on: {type: Date, optional: true},
+  created_on: {type: Date},
+  due_on: { type: Date },
```
Comment (Ligerx) [inline imports/api/tasks/tasks.js]: due_on is optional

File: imports/api/tasks/tasks.js [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+  is_urgent: { type: Boolean},
+  is_important: { type: Boolean}
```
Comment (Ligerx) [inline imports/api/tasks/tasks.js]: Can you default these both to false?
Hmmm not sure how to do this actually. How do we differentiate between unsorted and not urgent + not important? Using nulls feels kind of unsafe

File: imports/api/tasks/tasks.js (hunk 1 above)
Comment (Ligerx) [reply imports/api/tasks/tasks.js]: jk I think you need to type export {Tasks}

File: imports/api/tasks/tasks.js (hunk 3 above)
Comment (suvrathpen) [reply imports/api/tasks/tasks.js]: That's a good point. Should we have another boolean for now such as is_sorted

**Pull request merged**